messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistSdk.kt

```diff
@@ -38,20 +39,33 @@ class GistSdk(
 
     private var timer: Timer? = null
     private val gistQueue = SDKComponent.gistQueue
+    private val sseConnectionManager = SDKComponent.sseConnectionManager
 
     private fun resetTimer() {
         timer?.cancel()
         timer = null
     }
 
     private fun onActivityResumed() {
-        logger.debug("Activity resumed, starting polling")
-        fetchInAppMessages(state.pollInterval)
+        logger.debug("Activity resumed")
+        if (state.sseEnabled) {
+            logger.info("SSE is enabled, starting SSE connection")
+            sseConnectionManager.startConnection()
+        } else {
+            logger.debug("SSE is disabled, starting polling")
+            fetchInAppMessages(state.pollInterval)
+        }
     }
 
     private fun onActivityPaused() {
-        logger.debug("Activity paused, stopping polling")
-        resetTimer()
+        logger.debug("Activity paused")
+        if (state.sseEnabled) {
+            logger.info("SSE is enabled, stopping SSE connection")
+            sseConnectionManager.stopConnection()
+        } else {
+            logger.debug("SSE is disabled, stopping polling")
+            resetTimer()
+        }
```

Comment:
i see around 2-3 different places performing similar steps with few changes, should we probably have a function which takes in lambda?